homeassistant/components/modbus/climate.py

```diff
@@ -89,6 +98,28 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
     temp_step = config[CONF_STEP]
     hub_name = config[CONF_HUB]
     hub = hass.data[MODBUS_DOMAIN][hub_name]
+    structure = config[CONF_STRUCTURE]
+
+    if data_type != DATA_TYPE_CUSTOM:
+        try:
+            structure = f">{DEFAULT_STRUCT_FORMAT[data_type][count]}"
```

Comment:
Use the constant here as well.

Reply:
What part of the expression should I put into constant? Right now, a struct format is selected from an array based on data type and count values.

Reply:
Are we really asking the user to add something like “>f” (default for structure), would be nice if it was something simpler.

Reply:
@janiversen, `CONF_STRUCTURE` is an optional parameter, which is not used under normal circumstances. In most cases, user gives `data_count` and `data_type` parameters, and based on them we select a proper structure matching data type and number of bytes.

We list only few common types there, and sometimes data aren't parsed correctly. Therefore the `struct` option. Users who use this option, should be well aware of what they're doing, and they're given full control. From my point of view, this is acceptable.

Reply:
I suspected that, I will take a closer look at the documentation.